resqpy/model/_hdf5.py

```diff
@@ -0,0 +1,344 @@
+    return h5_root[h5_key_pair[1]][slice_tuple]
+
+
+def _h5_overwrite_array_slice(model, h5_key_pair, slice_tuple, array_slice):
+    """Overwrites (updates) a slice of an hdf5 array."""
+
+    h5_root = _h5_access(model, h5_key_pair[0], mode = 'a')
+    dset = h5_root[h5_key_pair[1]]
+    dset[slice_tuple] = array_slice
+
+
+def h5_clear_filename_cache(model):
+    """Clears the cached filenames associated with all ext uuids."""
+
+    _h5_release(model)
+    model.h5_dict = {}
+
+
+def _change_hdf5_uuid_in_hdf5_references(model, node, old_uuid, new_uuid):
+    """Scan node for hdf5 references and set the uuid of the hdf5 file itself to new_uuid."""
+
+    count = 0
+    old_uuid_str = str(old_uuid)
+    new_uuid_str = str(new_uuid)
+    for ref_node in node.iter(ns['eml'] + 'HdfProxy'):
+        try:
+            uuid_node = rqet.find_tag(ref_node, 'UUID')
+            if old_uuid is None or uuid_node.text == old_uuid_str:
+                uuid_node.text = new_uuid_str
+                count += 1
+        except Exception:
+            pass
+    if count == 1:
+        log.debug('one hdf5 reference modified')
+    else:
+        log.debug(str(count) + ' hdf5 references modified')
+    if count > 0:
+        model.set_modified()
+
+
+def _change_uuid_in_hdf5_references(model, node, old_uuid, new_uuid):
+    """Scan node for hdf5 references using the old_uuid and replace with the new_uuid."""
+
+    count = 0
+    old_uuid_str = str(old_uuid)
+    new_uuid_str = str(new_uuid)
+    for ref_node in node.iter(ns['eml'] + 'PathInHdfFile'):
+        try:
+            uuid_place = ref_node.text.index(old_uuid_str)
+            new_path_in_hdf = ref_node.text[:uuid_place] + new_uuid_str + ref_node.text[uuid_place + len(old_uuid_str):]
+            log.debug('path in hdf update from: ' + ref_node.text + ' to: ' + new_path_in_hdf)
+            ref_node.text = new_path_in_hdf
+            count += 1
+        except Exception:
+            pass
+    if count == 1:
+        log.debug('one hdf5 reference modified')
+    else:
+        log.debug(str(count) + ' hdf5 references modified')
+    if count > 0:
+        model.set_modified()
+
+
+def _change_filename_in_hdf5_rels(model, new_hdf5_filename = None):
+    """Scan relationships forest for hdf5 external parts and patch in a new filename."""
+
+    if not new_hdf5_filename and model.epc_file and model.epc_file.endswith('.epc'):
+        new_hdf5_filename = os.path.split(model.epc_file)[1][:-4] + '.h5'
+    count = 0
+    for rel_name, entry in model.rels_forest.items():
+        rel_root = entry[1].getroot()
+        for child in rel_root:
+            if child.attrib['Id'] == 'Hdf5File' and child.attrib['TargetMode'] == 'External':
+                child.attrib['Target'] = new_hdf5_filename
+                count += 1
+    log.info(str(count) + ' hdf5 filename' + _pl(count) + ' set to: ' + new_hdf5_filename)
+    if count > 0:
+        model.set_modified()
+
+
+def _pl(i, e = False):
```

Comment:
Function is duplicated in resqpy\well\well_utils. Do we need both versions?

Reply:
I guess we should have just a single copy in some obscure olio module. It's such a tiny function I simple cut and pasted it, probably in a few places!